frontend/src/app/modules/new-meeting/new-meeting/new-meeting.component.ts

```diff
@@ -0,0 +1,207 @@
+
+    public duration: number;
+
+    public customTimeShown: boolean = false;
+
+    public mainContentCustomTimeShown: boolean = false;
+
+    public meetingForm: FormGroup;
+
+    public memberFilterCtrl: FormControl = new FormControl();
+
+    public meetingNameControl: FormControl = new FormControl('', [
+        Validators.required,
+        Validators.minLength(2),
+        Validators.maxLength(50),
+        Validators.pattern(newMeetingNameRegex),
+    ]);
+
+    public customTimeControl: FormControl = new FormControl('', [
+        Validators.pattern(naturalNumberRegex),
+    ]);
+
+    public mainContainerCustomTimeControl: FormControl = new FormControl('', [
+        Validators.pattern(naturalNumberRegex),
+    ]);
+
+    ngOnInit(): void {
+        this.meetingForm = new FormGroup({
+            meetingName: this.meetingNameControl,
+            customTime: this.customTimeControl,
+            unitOfTime: new FormControl(),
+            location: new FormControl(),
+            duration: new FormControl(),
+            mainContainerDuration: new FormControl(),
+            mainContainerCustomTime: this.mainContainerCustomTimeControl,
+            mainContainerUnitOfTime: new FormControl(),
+            date: new FormControl('', [Validators.required]),
+            teamMember: new FormControl(),
+        });
+
+        this.patchFormValues();
+        this.setValidation();
+    }
+
+    public create(form: FormGroup) {
+        if (this.meetingForm.valid) {
+            const newMeeting: INewMeeting = {
+                name: form.value.meetingName,
+                location: form.value.location,
+                duration: this.duration,
+                startTime: form.value.date,
+                meetingLink: form.value.meetingName,
+                meetingMembers: this.addedMembers,
+                createdAt: new Date(),
+            };
+
+            this.newMeetingService.saveNewMeeting(newMeeting)
+                .pipe(this.untilThis)
+                .subscribe(() => {
+                    this.notificationService.showSuccessMessage('New meeting was created successfully.');
+                    this.reset();
+                });
+        } else {
+            this.notificationService.showErrorMessage('All fiels need to be set');
+        }
+    }
+
+    public getTeamMembersOfCurrentUser() {
+        this.newMeetingService
+            .getTeamMembersOfCurrentUser()
+            .pipe(this.untilThis)
+            .subscribe((resp) => {
+                this.teamMembers = resp;
+
+                this.filteredOptions = this.memberFilterCtrl.valueChanges.pipe(
+                    startWith(''),
+                    map(value => {
+                        const name = typeof value === 'string' ? value : value?.name;
+
+                        return name ? this._filter(name as string) : this.teamMembers.slice();
+                    }),
+                );
```

Comment:
Move this to a separate method
This logic looks a little bit strange try to refactor this

Reply:
Done.